sunpy/database/database.py

```diff
@@ -354,20 +363,26 @@ def download(self, *query, **kwargs):
         """
         if not query:
             raise TypeError('at least one attribute required')
-        client = kwargs.pop('client', None)
-        path = kwargs.pop('path', None)
-        progress = kwargs.pop('progress', False)
-        if kwargs:
-            k, v = kwargs.popitem()
+        
+        kwargs_copy = kwargs.copy()
```

Comment:
@dpshelio A copy is made because first I'm checking if any unsupported arguments are passed by popping out client, path, progress and methods. In this step `kwargs_copy` is emptied.
```
if kwargs_copy:
            k, v = kwargs_copy.popitem()
            raise TypeError('unexpected keyword argument {0!r}'.format(k))
```
After that the entire `kwargs` needs to be passed to the `_download_and_collect_entries` function.